File: app/src/main/java/com/coppermobile/mysamplemvvmdatabindinglivedata/data/source/DataSource.java
package com.coppermobile.mysamplemvvmdatabindinglivedata.data.source;

import android.arch.lifecycle.LiveData;

import com.coppermobile.mysamplemvvmdatabindinglivedata.data.Comments;
import com.coppermobile.mysamplemvvmdatabindinglivedata.data.Dish;

import java.util.List;

public interface DataSource<T> {

    void insertData(T data);

    void insertData(List<T> data);

    void deleteData(T data);

    LiveData<List<Comments>> getComments(int id);

    LiveData<List<Dish>> getAllDishes();
}

File: app/src/main/java/com/coppermobile/kotlinsamplemvvmlivedataroom/data/source/DataSource.kt
package com.coppermobile.kotlinsamplemvvmlivedataroom.data.source

import android.arch.lifecycle.LiveData
import com.coppermobile.kotlinsamplemvvmlivedataroom.data.Comments
import com.coppermobile.kotlinsamplemvvmlivedataroom.data.Dish

interface DataSource<T> {

    fun insertData(data: T)

    fun insertData(data: List<T>)

    fun deleteData(data: T)

    fun getComments(id: Int): LiveData<List<Comments>>?

    fun getAllDishes(): LiveData<List<Dish>>
}